lib/Secrets/doc/sailfish-secrets-overview.qdoc

```diff
@@ -0,0 +1,242 @@
+/****************************************************************************************
+**
+** Copyright (C) 2018 Jolla Ltd.
+** Contact: Chris Adams <chris.adams@jollamobile.com>
+** All rights reserved.
+**
+****************************************************************************************/
+
+/*!
+\contentspage {Sailfish OS Secrets Library Contents}
+\page sailfish-secrets-overview.html
+
+\title Sailfish OS Secrets Library Overview
+
+The Sailfish OS Secrets Library provides applications with API to access the
+secure storage functionality offered by the system service daemon as part of
+the Sailfish OS Secrets and Crypto Framework. Please see the Sailfish OS
+Secrets and Crypto Framework documentation for more in-depth documentation
+about the framework architecture, and broad overview of the features the
+framework provides.
+
+\section1 Using the Sailfish OS Secrets Library
+
+As described in the Sailfish OS Secrets and Crypto Framework overview
+documentation, client applications can use the Sailfish OS Secrets Library
+in order to make use of the secure storage services provided by the
+Sailfish OS Secrets and Crypto Framework.
+
+This library provides client applications written in C++ (with Qt) with API to
+make use of the secure storage services provided by the Sailfish OS Secrets and
+Crypto Framework.
+
+To make use of this library, applications should depend on the
+"sailfishsecrets.pc" pkgconfig file.
+
+e.g. in a qmake-based project:
+
+\code
+CONFIG += link_pkgconfig
+PKGCONFIG += sailfishsecrets
+\endcode
+
+\section2 Client API
+
+The client API consists of a variety of C++ classes which represent the
+inputs to secure storage operations (including secrets of various types),
+the result of a secure storage operation (that is, whether it succeeded or
+failed, along with some information about the reason for the failure),
+and one class which provides an interface to the remote service.
+
+\list
+\li \l{Sailfish::Secrets::Secret} represents a (possibly partial or reference) secret
+\li \l{Sailfish::Secrets::Secret::Identifier} consists of a secret name and optionally a collection name
+\li \l{Sailfish::Secrets::Result} represents the result (but not the output) of a secure storage operation
+\li \l{Sailfish::Secrets::SecretManager} provides an interface to the system secure storage service
+\endlist
+
+There also exist some API for dealing with authentication flows which require
+user interaction, where that user interaction is not mediated by the system
+(e.g., device lock). These APIs are listed here only for the sake of
+completeness as most applications should never use them:
```

Comment:
I find this useful, but I guess it is noise for most app developers. Maybe it is good to be inclusive in overview documentation.

Reply:
I'd prefer not to leave out details, since some people might look at the header list and think "but what about these exported classes?  what are they for?  do I need them?" -- better to clearly state that they are not useful in the majority of circumstances, and let the developer move on, IMO.